src/components/button/styled-components.js

```diff
@@ -1,30 +1,38 @@
 /* eslint-disable flowtype/require-valid-file-annotation */
 import {styled} from '../../styles';
-import {BUTTON_TYPE} from './constants';
+import {BUTTON_KIND, BUTTON_SIZE} from './constants';
 
-export const BaseButton = styled('button', ({$theme, $type}) => ({
-  ...$theme.typography.font300,
+export const BaseButton = styled('button', ({$theme, $size, $kind}) => ({
   display: 'flex',
+  ...$theme.typography[$size === BUTTON_SIZE.compact ? 'font200' : 'font300'],
   alignItems: 'center',
   justifyContent: 'space-between',
   border: 'none',
-  borderRadius: $type === BUTTON_TYPE.round ? '50%' : $theme.borders.radius200,
+  borderRadius: $kind === BUTTON_KIND.round ? '50%' : $theme.borders.radius200,
   outline: 'none',
   WebkitAppearance: 'none',
+  transitionDuration: $theme.animation.timing100,
+  transitionTimingFunction: $theme.animation.easeOutCurve,
```

Comment:
Should we be explicit about which properties are being transitioned? (background color, etc).

Reply:
Good point, I think it should only be the background color. I thought the immediate change of background color looks too harsh.